// angular2-image-gallery/esm2020/lib/angular2-image-gallery.module.mjs
import { NgModule } from '@angular/core';
import { ImageService } from './services/image.service';
import { GalleryComponent } from './gallery/gallery.component';
import { ViewerComponent } from './viewer/viewer.component';
import { CommonModule } from '@angular/common';
import 'hammerjs';
import * as i0 from "@angular/core";
export class Angular2ImageGalleryModule {
}
Angular2ImageGalleryModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: Angular2ImageGalleryModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
Angular2ImageGalleryModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: Angular2ImageGalleryModule, declarations: [GalleryComponent, ViewerComponent], imports: [CommonModule], exports: [GalleryComponent, ViewerComponent] });
Angular2ImageGalleryModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: Angular2ImageGalleryModule, providers: [ImageService], imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: Angular2ImageGalleryModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [GalleryComponent, ViewerComponent],
                    providers: [ImageService],
                    exports: [GalleryComponent, ViewerComponent],
                }]
        }] });
